src/main/java/com/moodybluez/enterprise/controller/EntryController.java

```diff
@@ -27,15 +24,17 @@ public Entry getById(@PathVariable int id) {
         List<Entry> entries = entryService.fetchByMonth(year, month);
         Map<Integer, Entry> ret = new HashMap<>();
         for(Entry entry: entries){
-            ret.put(entry.getDate().getDay(),entry);
+            Calendar calendarInstance = Calendar.getInstance();
+            calendarInstance.setTime(entry.getDate());
+            int dayOfWeek = calendarInstance.get(Calendar.DAY_OF_WEEK);
+            ret.put(dayOfWeek,entry);
         }
         return ret;
     }
 
     @GetMapping("entry/{year}/{month}/{day}")
     Entry getByMonth(@PathVariable Integer year, @PathVariable Integer month, @PathVariable Integer day) {
-        Entry entry = entryService.fetchByDate(year.toString()+"-"+month.toString()+"-"+day.toString());
-        return entry;
+        return entryService.fetchByDate(year.toString()+"-"+month.toString()+"-"+day.toString());
```

Comment:
No need to assign to variable and then return that variable, especially you are not either using that variable elsewhere or  surrounding it with try-catch block. This is not a big issue here but could slow down application if there was a lot going on here and if this practice was repeated.